daemon/containerd/image_tag.go

```diff
@@ -42,8 +56,10 @@ func (i *ImageService) TagImage(ctx context.Context, imageID image.ID, newTag re
 
 		// Check if image we would replace already resolves to the same target.
 		// No need to do anything.
-		if replacedImg.Target.Digest == targetImage.Target.Digest {
-			i.LogImageEvent(imageID.String(), reference.FamiliarString(newTag), events.ActionTag)
+		if replacedImg.Target.Digest == newImg.Target.Digest {
+			if !creatingDangling {
+				i.LogImageEvent(replacedImg.Target.Digest.String(), imageFamiliarName(newImg), events.ActionTag)
```

Comment:
Note to self; this one  will conflict with a PR I had pending (I need to update that one, but it's less important  than this PR);

- https://github.com/moby/moby/pull/49014